SEP-0008.md

```diff
@@ -0,0 +1,179 @@
+    be denoted using .devM where M is a non-negative integer number. Development
+    releases are ordered by their numerical component, immediately before the
+    corresponding release (and before any pre-releases with the same release
+    segment), and following any previous release (including any post-releases).
+    Development releases MAY also be used in conjunction with pre-release
+    numbering.
+    Examples: 1.0.0.dev25, 1.2.3a1.dev4532 1.0.0rc2.dev3456
+
+1.  Precedence refers to how versions are compared to each other when
+    ordered. Precedence MUST be calculated by separating the version
+    into major, minor, patch and pre-release identifiers in that order.
+    Precedence is
+    determined by the first difference when comparing each of these
+    identifiers from left to right as follows: Major, minor, and patch
+    versions are always compared numerically. Example: 1.0.0 < 2.0.0
+    < 2.1.0 < 2.1.1. When major, minor, and patch are equal, a
+    pre-release version has lower precedence than a normal version. Example:
+    1.0.0a1 < 1.0.0. Precedence for two pre-release versions with the
+    same major, minor, and patch version MUST be determined by following the
+    precedence rules for pre-release versioning. PEP 440 describes a formal way
+    of matching version numbers, which this specification is compatible with.
+    
+
+# Future SunPy Releases
+
+As this document is being written the 0.8.z release of SunPy is the last
+non-bugfix release. This section describes the releases to follow 0.8.z and how
+the above numbering specification will relate to future releases.
+
+0.8.z introduced a number of major deprecations (e.g. wcs and lightcurve), and
+0.9.z is scheduled to be the last release to support Python 2.7. As SunPy has
+reached a stage of substantial use and some submodules, such as map and net have
+reached a state of relative maturity, this SEP proposes that the release that
+follows 0.9.z will be the SunPy 1.0.0 release.
+
+Releasing 1.0.0, and following, the above version specification will define the
+public API of SunPy, this API will not be allowed to change without incrementing
+the major version number, i.e. 1.0.0 to 2.0.0. 
+
+## SunPy API Definition
+
+
+The SunPy public API will be defined by the documented behaviour (the content of
+the docstrings) of **any object that is present in the `__all__` list in the
+Python source, and any attributes on those objects not prefixed with one or more
+underscores**, and are therefore included in the API documentation.
+
+A change to the public API is anything that affects not just the name, arguments
+or properties of any object, but any change that alters the behaviour of that
+object (i.e. the values returned from calling it). Deprecating some behaviour is
+a backwards compatible change, and removing it is not backwards compatible.
+Where possible behaviour should be deprecated before it is removed.
+
+## SunPy Release Schedule
```

Comment:
This section mixes plans for SunPy's development and its versioning description. I recommend that this section be dropped.

Reply:
Maybe we should have a separate SEP for describing how and when we release.  For example, I'd be OK with a new patch release every time a single new patch was implemented.  I think when a user finds a bug we should be as responsive as possible, otherwise we could be blocking someone's research.

Reply:
Agreed.